pkg/proxy/proxy.go

```diff
@@ -63,7 +63,10 @@ func (p *SprayProxy) HandleProxy(c *gin.Context) {
 	}
 	body := buf.Bytes()
 
-	client := &http.Client{}
+	client := &http.Client{
+		// set forwarding request timeout
+		Timeout: 15 * time.Second,
```

Comment:
feels like the amount of time here should be configurable

Reply:
Any preference how the value should be set? I mean command line param, env var or add a config file? Or any constellation of those?

Reply:
I don't have a strong preference, but if we think we'll be tuning this more than other items, and env var is a bit easier from a gitops perspective

unlike some of the command arg stuff that already exists, I could see this being a setting we tweak on the fly in prod as part of "providing support" i.e. we need to make the timeout longer because there is more load on the system

wrt a configmap/configfile, I at least am fine waiting on that until we reach some TBD "critical mass" on how many config options we expose.

Reply:
Introduced SPRAYPROXY_FORWARDING_REQUEST_TIMEOUT env var which can be used to override the default timeout. It's only read once on server start, so to take effect the server must be restarted.
I did not want to handle it with viper as the value has to be passed as a parameter to two other "constructors" which already get a significant number of parameters. As part of a future refactoring, we can move the configuration management to a separate pkg which can help reducing the number of parameters passed around.